File: frontend/src/lib/services/statistics.ts
import { api } from "../api-client";

/**
 * 统计契约(API.md §7)。by_status 为稀疏键:某状态计数为 0 时键可省略,
 * 前端必须区分「零」「缺失」与「不可用」;指标族整体也可能缺失。
 */

export type StatisticsScope = "personal" | "team" | "organization" | "system";

export type RecordFamilyStats = {
  total: number;
  /** 稀疏键:仅包含有数据的状态。 */
  by_status: Record<string, number>;
};

export type TaskStats = {
  total: number;
  by_status: Record<string, number>;
};

export type KnowledgeStats = {
  document_count: number;
  indexed_count: number;
  indexing_count: number;
  failed_count: number;
};

export type Statistics = {
  scope: StatisticsScope;
  generated_at: string;
  records: {
    inspection_records?: RecordFamilyStats;
    photo_reports?: RecordFamilyStats;
    interview_records?: RecordFamilyStats;
  };
  tasks?: TaskStats;
  knowledge?: KnowledgeStats;
  generated_documents?: { total: number };
};

export const statisticsService = {
  get: (signal?: AbortSignal) => api.get<Statistics>("/api/statistics", { signal }),
};

File: backend/app/schemas/statistics.py
"""Statistics schema（API.md §7）。"""

from datetime import datetime

from pydantic import BaseModel


class StatusCount(BaseModel):
    total: int
    by_status: dict[str, int]


class RecordsStats(BaseModel):
    inspection_records: StatusCount
    photo_reports: StatusCount
    interview_records: StatusCount


class KnowledgeStats(BaseModel):
    """知识库聚合计数（API.md §7 knowledge 段）。"""

    document_count: int = 0
    indexed_count: int = 0
    indexing_count: int = 0
    failed_count: int = 0


class GeneratedDocumentsStats(BaseModel):
    total: int


class StatisticsResponse(BaseModel):
    scope: str
    generated_at: datetime
    records: RecordsStats
    tasks: StatusCount
    knowledge: KnowledgeStats
    generated_documents: GeneratedDocumentsStats
